pkg/api/validation/validation.go

```diff
@@ -2444,6 +2421,8 @@ func validatePodAffinityTerm(podAffinityTerm api.PodAffinityTerm, allowEmptyTopo
 	}
 	if len(podAffinityTerm.TopologyKey) != 0 {
 		allErrs = append(allErrs, unversionedvalidation.ValidateLabelName(podAffinityTerm.TopologyKey, fldPath.Child("topologyKey"))...)
+	} else {
```

Comment:
Regarding test cases, the idea was that if affinity in annotation is disabled then, topology key length should not be zero and thats why that test cases. I have only been saying that if you are adding  len(podAffinityTerm.TopologyKey) != 0 , then please remove `if !allowEmptyTopologyKey && len(podAffinityTerm.TopologyKey) == 0 {`.